<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebugWindow</class>
 <widget class="QMainWindow" name="DebugWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1107</width>
    <height>715</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::TabPosition::South</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::TabShape::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabExpressions">
       <attribute name="title">
        <string>Expressions</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="Debugger::ExpressionsView" name="tableViewExpressions">
          <property name="gridStyle">
           <enum>Qt::PenStyle::DashLine</enum>
          </property>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxRunCode">
          <property name="title">
           <string>Run code</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLineEdit" name="lineEditCodeToRun"/>
           </item>
           <item>
            <widget class="QPushButton" name="pushButtonRunCode">
             <property name="text">
              <string>Run</string>
             </property>
             <property name="icon">
              <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabLog">
       <attribute name="title">
        <string>Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPlainTextEdit" name="logText"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1107</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuDebug_Log">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuLog">
    <property name="title">
     <string>Log</string>
    </property>
    <addaction name="actionLogClear"/>
    <addaction name="actionLogCopyAll"/>
   </widget>
   <widget class="QMenu" name="menuExpressions">
    <property name="title">
     <string>Expressions</string>
    </property>
    <addaction name="actionExpressionAdd"/>
    <addaction name="actionExpressionRemove"/>
    <addaction name="actionExpressionRemoveAll"/>
    <addaction name="actionExpressionCopy"/>
    <addaction name="actionExpressionCopyAll"/>
   </widget>
   <addaction name="menuDebug_Log"/>
   <addaction name="menuExpressions"/>
   <addaction name="menuLog"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionExpressionAdd"/>
   <addaction name="actionRefreshExpressions"/>
   <addaction name="actionActionAutoRefreshExpressions"/>
   <addaction name="separator"/>
   <addaction name="actionLogClear"/>
   <addaction name="actionLogCopyAll"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionLogClear">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditClear"/>
   </property>
   <property name="text">
    <string>&amp;Clear</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset theme="view-close"/>
   </property>
   <property name="text">
    <string>C&amp;lose</string>
   </property>
  </action>
  <action name="actionLogCopyAll">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditCopy"/>
   </property>
   <property name="text">
    <string>Copy all</string>
   </property>
  </action>
  <action name="actionRefreshExpressions">
   <property name="icon">
    <iconset theme="view-refresh"/>
   </property>
   <property name="text">
    <string>Refresh expressions</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionActionAutoRefreshExpressions">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="folder-sync"/>
   </property>
   <property name="text">
    <string>actionAutoRefreshExpressions</string>
   </property>
   <property name="toolTip">
    <string>Automatically refresh expression values</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionExpressionAdd">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::ListAdd"/>
   </property>
   <property name="text">
    <string>Add</string>
   </property>
   <property name="shortcut">
    <string>Ins</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionExpressionRemove">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::ListRemove"/>
   </property>
   <property name="text">
    <string>Remove</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionExpressionCopy">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditCopy"/>
   </property>
   <property name="text">
    <string>Copy value</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionExpressionCopyAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Copy All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actionExpressionRemoveAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Remove All</string>
   </property>
   <property name="shortcut">
    <string>Shift+Del</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Debugger::ExpressionsView</class>
   <extends>QTableView</extends>
   <header>expressionsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
